### DISCUSSION TITLE
'brew cleanup' fails with 'Permission denied' issue 'Error: Permission denied @ apply2files'

### DISCUSSION THREAD

**Discussion by nairs** [Homebrew/discussions#651] [Everyday usage]
Notice this issue was an earlier reported and closed under "brew cleanup: Error: Permission denied @ apply2files #45009". I am still getting this problem - hence creating this new thread. 

What I found out - brew cleanup - is trying to clean folders created by npm. As you know, npm allows run as root with 'sudo' and brew is not allowing sudo. 

May be only for cleanup - we should consider allowing sudo. 

Also, we need to check, how to limit brew cleanup to packages installed by brew - and brew should not try to cleanup - packages installed by other package managers like npm.

**Comment by SMillerDev** [comment]
You really shouldn't run npm as root either. There have been multiple bugs where that removed system files and there is no real benefit.

**Comment by gromgit** [comment]
When various folks over the years have written about [how to avoid using sudo when installing packages globally](https://medium.com/@ExplosionPills/dont-use-sudo-with-npm-still-66e609f5f92), and even npm's docs [agrees with them](https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally/), I think it's safe to say that Homebrew won't support sudo usage for this.
